src/components/Interval.js

```diff
@@ -0,0 +1,62 @@
+import { Component } from 'react'
+import renderProps from '../utils/renderProps'
+
+class Interval extends Component {
+  static defaultProps = {
+    delay: 1000,
+  }
+
+  intervalId = undefined
+
+  _clearIntervalIfNecessary = () => {
+    if (this.intervalId) {
+      this.intervalId = clearInterval(this.intervalId)
+    }
+  }
+
+  _setIntervalIfNecessary = delay => {
+    if (Number.isFinite(delay)) {
+      this.intervalId = setInterval(() => this.forceUpdate(), delay)
```

Comment:
Bad thing. `forceUpdate` cannot be optimised with react. It's better to `setState` incrementing interval value.

Reply:
Humm, nice to know